packages/core/src/telemetry/detached-flush.js

```diff
@@ -19,13 +19,15 @@ async function detachedFlush() {
     events.map(async (event) => {
       await fetch(telemetryUrl, {
         method: "POST",
-        body: JSON.stringify(event.payload),
+        body: JSON.stringify(event),
         headers: {
           "Content-Type": "application/json",
         },
       });
     })
-  );
+  ).finally(() => {
+    fs.rmSync(eventsFilePath);
+  });
```

Comment:
I would probably change this to
```ts
try {
  await Promise.all( ... )
} catch (error) {
  console.error(error)
}

fs.rmSync(eventsFilePath)
```

[See this](https://stackoverflow.com/a/58056140).

Reply:
Hmm - I think it's fine to ignore any errors coming from the script tbh. The only real source of errors is segment HTTP request failure/error. Our test for the script is pretty thorough.